parsers/CO.py

```diff
@@ -91,19 +97,22 @@ def fetch_historical_consumption(
 ) -> list[dict[str, Any]]:
     demand_list = TotalConsumptionList(logger)
     # Convert datetime to local time
-    target_arrow_in_tz = arrow.get(target_datetime).to(TZ)
+    target_datetime_in_tz = target_datetime.astimezone(tz=TZ)
```

Comment:
Target datetime might be `None` here and we should therefor not call `.astimezone()` directly here.

We should do something like this instead:
```python
    now = datetime.now(tz=TIMEZONE)
    target_datetime_in_tz = (
         now if target_datetime is None else target_datetime.astimezone(TIMEZONE)
     )
```